### PR TITLE
feat(state-viewer): list accounts with contracts

### PR DESCRIPTION
**Pull request by jakmeier**
A state viewer command to list all accounts with contracts deployed.

So far it only shows the account names and the wasm size.
The intention is to extend it for more sophisticated filtering.

So far, the command finishes all four shards in 2 minutes on a 
mainnet archival RocksDB.

### CODE REVIEW

File: tools/state-viewer/src/contract_accounts.rs
Comment (pugachAG) [inline tools/state-viewer/src/contract_accounts.rs:75]: just fyi: I'm working on exposing API around trie key ranges as part of #8332, I will try to keep this use case in mind when designing it, so we can replace this with something nicer.

File: tools/state-viewer/src/contract_accounts.rs
Comment (pugachAG) [inline tools/state-viewer/src/contract_accounts.rs:92]: optional suggestion/nit:
```
while let Some(item) = self.contract_nodes.pop_front() {
    if let Some(TrieTraversalItem { hash, key: Some(trie_key) }) = item {
       ...
    }
}
```
To me this seems to be a bit more easier to read, no strong opinion though, up to you :)

File: tools/state-viewer/src/cli.rs
Comment (jakmeier) [reply tools/state-viewer/src/cli.rs]: oops, copy-pasta error from a different branch I am working on ^^

File: tools/state-viewer/src/contract_accounts.rs
Comment (jakmeier) [reply tools/state-viewer/src/contract_accounts.rs:75]: good to know, thanks

File: tools/state-viewer/src/contract_accounts.rs
Comment (jakmeier) [reply tools/state-viewer/src/contract_accounts.rs]: oh yeah, I like it, thanks!

File: tools/state-viewer/src/commands.rs
Comment (jakmeier) [reply tools/state-viewer/src/commands.rs:856]: I can see the consistency argument but I believe `eprintln` is the right choice here anyway.

The reason this goes to stderr is precisely because I don't want it in redirected output. This is just a general progress update to reassure whoever sits in front of the terminal that things are moving along. But scripts working with the output will not want that line.

Case in point, in [this comment](https://github.com/near/nearcore-private/issues/1#issuecomment-1384567730) I have outlined a number of in-terminal data transformations. The script below that adds up all sizes would actually break if there was a line like the above, where the second field is not a number.

```bash
awk '{total+=$2} END{printf "%d\n", total}' < contract_accounts.log 
```

Comment (jakmeier) [reply tools/state-viewer/src/commands.rs:856]: Oh but if you still think it's better for consistency to not write to `eprintln!`, then I can also just remove this line, I don't care about it that much. As long as it doesn't go to stdout. :)

Comment (pugachAG) [reply tools/state-viewer/src/commands.rs:856]: I see your point and it definitely makes sense to print progress messages to stderr.

Let's keep `eprintln!` here, I think the right solution would be to fix all other instances of printing progress messages to stdout (but definitely not a part of this PR).

**Review by pugachAG — APPROVED** [review]
LGTM!

**Comment by jakmeier** [comment]
Thanks for the review.

@Longarithm I will also need a code owner approval on this one, pretty please :)

File: tools/state-viewer/src/contract_accounts.rs
Comment (Longarithm) [inline tools/state-viewer/src/contract_accounts.rs:65]: This doesn't look good, and this is a cause of long-standing annoying impl detail of `TrieKey`.
I wrote an issue about it: https://github.com/near/nearcore/issues/8376. Could you add TODO with this issue as a comment here?

**Review by Longarithm — APPROVED** [review]

File: tools/state-viewer/src/contract_accounts.rs
Comment (jakmeier) [reply tools/state-viewer/src/contract_accounts.rs:65]: Thanks for filing the issue! Yes I will add the TODO comment

**Pull request merged**
